src/operator/controllers/intents_reconcilers/otterizecloud/status_report.go

```diff
@@ -0,0 +1,38 @@
+package otterizecloud
+
+import (
+	"context"
+	"github.com/otterize/intents-operator/src/shared/otterizecloud/graphqlclient"
+	"github.com/otterize/intents-operator/src/shared/otterizecloud/otterizecloudclient"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+func PeriodicallyReportConnectionToCloud(client CloudClient) {
+	interval := viper.GetInt(otterizecloudclient.ComponentReportIntervalKey)
+	go func() {
+		runPeriodicReportConnection(interval, client)
+	}()
+}
+
+func runPeriodicReportConnection(interval int, client CloudClient) {
+	ctx, cloudClientCancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
```

Comment:
I'm not sure this will work, since there's already a signal handler in main (`mgr.Start(ctrl.SetupSignalHandler())`)
You should reuse the context that returns from `ctrl.SetupSignalHandler()`